gnucash/gnome/window-reconcile.c

```diff
@@ -2243,6 +2243,9 @@ recnFinishCB (GtkAction *action, RecnWindow *recnData)
     xaccAccountClearReconcilePostpone (account);
     xaccAccountSetReconcileLastDate (account, date);
 
+    xaccAccountSetBalanceAssertionAtDate (account, date,
+                                          recnData->new_ending);
```

Comment:
instead of `recnData->new_ending` we may need to use `gnc_time64_get_day_end(recnData->new_ending)`